superduperdb/components/datatype.py

```diff
@@ -104,17 +111,35 @@ class DataType(Component):
     decoder: t.Callable = dill_decode
     info: t.Optional[t.Dict] = None
     shape: t.Optional[t.Sequence] = None
-    artifact: bool = False
-    reference: bool = False
     directory: t.Optional[str] = None
-    encodable: t.Optional[str] = None
+    encodable: str = 'encodable'
+    bytes_encoding: str = CFG.bytes_encoding
+
+    def __post_init__(self, artifacts):
+        super().__post_init__(artifacts)
+        self.encodable_cls = _ENCODABLES[self.encodable]
+        self._takes_x = 'x' in inspect.signature(self.encodable_cls.__init__).parameters
```

Comment:
self.encodable._cls. some flag which tells us if its takes just uri or both x and uri 
Better than signature checking, WDYT? @blythed